extension/src/plots/model/index.ts

```diff
@@ -498,26 +458,21 @@ export class PlotsModel extends ModelWithPersistence {
     return this.commitRevisions[label] || label
   }
 
-  private getPlots(
-    checkpointPlots: CheckpointPlot[],
-    selectedExperiments: string[]
-  ) {
-    return reorderObjectList<CheckpointPlotData>(
-      this.metricOrder,
-      checkpointPlots.map(plot => {
-        const { id, values } = plot
-        return {
-          id,
-          title: truncateVerticalTitle(
-            id,
-            this.getNbItemsPerRow(Section.CHECKPOINT_PLOTS)
-          ) as string,
-          values: values.filter(value =>
-            selectedExperiments.includes(value.group)
-          )
-        }
-      }),
-      'id'
+  private getCustomPlotsData(
+    plots: CustomPlot[],
+    colors: ColorScale | undefined
+  ): CustomPlotData[] {
+    const selectedExperimentsExist = !!colors
+    let filteredPlots = plots
+    if (!selectedExperimentsExist) {
+      filteredPlots = plots.filter(plot => !isCheckpointPlot(plot))
+    }
+    return filteredPlots.map(plot =>
+      collectCustomPlotData(
```

Comment:
[C] What I meant here was that you don't have to filter and then map you could do it in a single loop:

```
  private getCustomPlotsData(
    plots: CustomPlot[],
    colors: ColorScale | undefined
  ): CustomPlotData[] {
    const selectedExperimentsExist = !!colors
    const filteredPlots: CustomPlotData[] = []
    for (const plot of plots) {
      if (!selectedExperimentsExist && isCheckpointPlot(plot)) {
        continue
      }
      filteredPlots.push(
        collectCustomPlotData(
          plot,
          colors,
          this.getNbItemsPerRow(Section.CUSTOM_PLOTS)
        )
      )
    }
    return filteredPlots
  }
```

```

Reply:
Fixed, thanks for the clarification!